### PR TITLE
drivers: espi: xec: Add support for eSPI OOB channel operations

### PR DESCRIPTION
**Pull request by albertofloyd**
Add support for eSPI OOB channel operations

Implement send/receive out-of-band packet APIs
Update eSPI sample driver with usage for eSPI OOB driver APIs

### CODE REVIEW

File: samples/drivers/espi/src/main.c [hunk 1]
```diff
@@ -31,7 +43,7 @@ static void espi_reset_handler(struct device *dev,
 			       struct espi_event event)
 {
 	if (event.evt_type == ESPI_BUS_RESET) {
-		printk("\neSPI BUS reset %d", event.evt_data);
+		LOG_INF("\neSPI BUS reset %d", event.evt_data);
```
Comment (nashif) [inline samples/drivers/espi/src/main.c]: Remove all newlines (\n), they come with LOG_* for free. Also below in all changes from printk to LOG_*...

**Review by nashif — CHANGES_REQUESTED** [review]

File: drivers/espi/espi_mchp_xec.c [hunk 2]
```diff
@@ -397,6 +406,72 @@ static int espi_xec_receive_vwire(struct device *dev,
...
+		LOG_WRN("OOB channel is disabled\n");
+		return -EIO;
+	}
+
+	if (ESPI_OOB_REGS->TX_STS & MCHP_ESPI_OOB_TX_STS_BUSY) {
+		LOG_WRN("OOB channel is busy\n");
+		return -EBUSY;
+	}
+
+	memcpy(&slave_tx_mem, pckt.buf, pckt.len);
```
Comment (dcpleung) [inline drivers/espi/espi_mchp_xec.c]: I think we should make sure that the packet can actually fit inside `slave_tx_mem`. Also remove `&`.

File: drivers/espi/espi_mchp_xec.c [hunk 3]
```diff
@@ -397,6 +406,72 @@ static int espi_xec_receive_vwire(struct device *dev,
...
+	}
+
+	/* Wait until ISR or timeout */
+	ret = k_sem_take(&data->rx_lock, MAX_OOB_TIMEOUT);
+	if (ret == -EAGAIN) {
+		return -ETIMEDOUT;
+	}
+
+	pckt.len = ESPI_OOB_REGS->RX_LEN;
+	memcpy(pckt.buf, slave_rx_mem, pckt.len);
```
Comment (dcpleung) [inline drivers/espi/espi_mchp_xec.c]: Need to check the `pckt.len >= RX_LEN`.

**Review by dcpleung — CHANGES_REQUESTED** [review]

**Review by franciscomunoz — COMMENTED** [review]
Address the current comments. I don't have anything new to add.

**Review by dcpleung — APPROVED** [review]

**Review by kashah1 — APPROVED** [review]

**Comment by albertofloyd** [comment]
@nashif  sample code was updated as per your comment

**Review by nashif — APPROVED** [review]

**Pull request merged**